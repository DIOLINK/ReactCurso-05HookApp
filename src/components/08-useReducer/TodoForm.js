import React from 'react';
import PropTypes from 'prop-types';
import useForm from '../../hooks/useForm';

const TodoForm = ({ handleAddTodo }) => {
  const [{ description }, handleInputChange, resetForm] = useForm({
    description: '',
  });
  const handleSutmit = (e) => {
    e.preventDefault();
    if (description.trim().length <= 1) {
      return;
    }

    const newTodo = {
      id: new Date().getTime(),
      description: description,
      done: false,
    };
    handleAddTodo(newTodo);
    resetForm();
  };
  return (
    <form onSubmit={handleSutmit}>
      <div className="d-grid gap-2">
        <input
          className="form-control form-control-lg"
          type="text"
          name="description"
          placeholder="Add TODO..."
          autoComplete="off"
          value={description}
          onChange={handleInputChange}
        />
        <button
          type="submit"
          className="btn btn-outline-primary mt-1 btn-block"
        >
          Add
        </button>
      </div>
    </form>
  );
};

TodoForm.propTypes = {
  handleAddTodo: PropTypes.func.isRequired,
};

export default TodoForm;
