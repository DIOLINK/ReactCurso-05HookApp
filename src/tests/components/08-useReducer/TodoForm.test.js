import { shallow } from 'enzyme';
import TodoForm from '../../../components/08-useReducer/TodoForm';

describe('Probando <TodoForm />', () => {
  const handleAddTodo = jest.fn();
  const wrapper = shallow(<TodoForm handleAddTodo={handleAddTodo} />);
  it('Debe de mostrarse correctamente', () => {
    expect(wrapper).toMatchSnapshot();
  });
  it('No debe de llamar a handleAddTodo', () => {
    const fomrSubmit = wrapper.find('form').prop('onSubmit');
    fomrSubmit({ preventDefault() {} });
    expect(handleAddTodo).toHaveBeenCalledTimes(0);
  });
  it('Debe de llamar a handleAddTodo', () => {
    const value = 'Aprender Jest';
    wrapper
      .find('input')
      .simulate('change', { target: { value, name: 'description' } });
    const fomrSubmit = wrapper.find('form').prop('onSubmit');
    fomrSubmit({ preventDefault() {} });
    expect(handleAddTodo).toHaveBeenCalledTimes(1);
    expect(handleAddTodo).toHaveBeenCalledWith({
      id: expect.any(Number),
      description: value,
      done: false,
    });
    expect(wrapper.find('input').prop('value')).toBe('');
  });
});
