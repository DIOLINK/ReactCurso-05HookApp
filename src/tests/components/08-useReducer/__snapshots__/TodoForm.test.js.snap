// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Probando <TodoForm /> Debe de mostrarse correctamente 1`] = `
<form
  onSubmit={[Function]}
>
  <div
    className="d-grid gap-2"
  >
    <input
      autoComplete="off"
      className="form-control form-control-lg"
      name="description"
      onChange={[Function]}
      placeholder="Add TODO..."
      type="text"
      value=""
    />
    <button
      className="btn btn-outline-primary mt-1 btn-block"
      type="submit"
    >
      Add
    </button>
  </div>
</form>
`;
